apps/src/netsim/NetSimRouterLogModal.js

```diff
@@ -267,18 +455,24 @@ NetSimRouterLogModal.prototype.setShard = function (newShard) {
     this.eventKeys_.registeredWithShard = newShard;
   }
 
+  // When changing shards, reset log so we fetch the whole thing next time.
+  this.logEntries_.length = 0;
+  this.latestRowID_ = 0;
   this.shard_ = newShard;
 };
 
 /**
  * Handle log table changes.
- * @param {LogEntryRow[]} logRows
  * @private
  */
-NetSimRouterLogModal.prototype.onLogTableChange_ = function (logRows) {
+NetSimRouterLogModal.prototype.onLogTableChange_ = function () {
   var headerSpec = NetSimGlobals.getLevelConfig().routerExpectsPacketHeader;
-  this.logEntries_ = logRows.map(function (row) {
+  var newRows = this.shard_.logTable.readAllFromID(this.latestRowID_ + 1);
+  var newLogEntries = newRows.map(function (row) {
+    this.latestRowID_ = Math.max(row.id, this.latestRowID_);
```

Comment:
reallAllFromID returns rows in sorted order, so I think can move this out of the loop and simply say

this.latestRowID_ = Math.max(this.latestRowID_, newRows[newRows.length - 1].id)